src/sentry/static/sentry/app/views/organizationStats/container.tsx

```diff
@@ -0,0 +1,314 @@
+        projectsRequestsPending: 1,
+        projectsLoading: true,
+      });
+    }
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    const prevParams = prevProps.params,
+      currentParams = this.props.params;
+
+    if (prevParams.orgId !== currentParams.orgId) {
+      this.fetchData();
+    }
+
+    // Query string is changed, probably due to pagination, re-fetch only project data
+    if (prevProps.location.search !== this.props.location.search) {
+      // Not sure why, but when we use pagination and the new results load and re-render,
+      // the scroll position does not reset to top like in Audit Log
+      if (window.scrollTo) {
+        window.scrollTo(0, 0);
+      }
+      this.fetchProjectData();
+    }
+    const state = this.state;
+    if (state.statsLoading && !state.statsRequestsPending) {
+      this.processOrgData();
+    }
+    if (state.projectsLoading && !state.projectsRequestsPending) {
+      this.processProjectData();
+    }
+  }
+
+  fetchProjectData() {
+    this.props.api.request(this.getOrganizationProjectsEndpoint(), {
+      query: this.props.location.query,
+      success: (data, _textStatus, jqxhr) => {
+        const projectMap: Record<string, Project> = {};
+        data.forEach((project: Project) => {
+          projectMap[project.id] = project;
+        });
+
+        this.setState(prevState => ({
+          pageLinks: jqxhr ? jqxhr.getResponseHeader('Link') : null,
+          projectMap,
+          projectsRequestsPending: prevState.projectsRequestsPending - 1,
+        }));
+      },
+      error: () => {
+        this.setState({
+          projectsError: true,
+        });
+      },
+    });
+  }
+
+  fetchData() {
+    this.setState({
+      statsError: false,
+      statsLoading: true,
+      statsRequestsPending: 3,
+      projectsError: false,
+      projectsLoading: true,
+      projectsRequestsPending: 4,
+    });
+
+    const statEndpoint = this.getOrganizationStatsEndpoint();
+
+    Object.keys(this.state.rawOrgData).forEach(statName => {
+      this.props.api.request(statEndpoint, {
+        query: {
+          since: this.state.querySince,
+          until: this.state.queryUntil,
+          resolution: '1h',
+          stat: statName,
+        },
+        success: data => {
+          this.setState(prevState => {
+            const rawOrgData = prevState.rawOrgData;
+            rawOrgData[statName] = data;
+
+            return {
+              rawOrgData,
+              statsRequestsPending: prevState.statsRequestsPending - 1,
+            };
+          });
+        },
+        error: () => {
+          this.setState({
+            statsError: true,
+          });
+        },
+      });
+    });
+
+    Object.keys(this.state.rawProjectData).forEach(statName => {
+      this.props.api.request(statEndpoint, {
+        query: {
+          since: this.state.querySince,
+          until: this.state.queryUntil,
+          stat: statName,
+          group: 'project',
+        },
```

Comment:
Will we be adding new endpoints that can allow the client to be less chatty with the server?

Reply:
yes, the page will only need to make two requests per time filter -- one to get overall org stats and one to get stats for projects the user belongs to